timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/DiskNamespaceLoader.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.management;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.slf4j.LoggerFactory;
+
+import com.palantir.atlasdb.timelock.paxos.PaxosUseCase;
+import com.palantir.logsafe.SafeArg;
+
+final class DiskNamespaceLoader {
+    private final Path rootDataDirectory;
+
+    DiskNamespaceLoader(Path rootDataDirectory) {
+        this.rootDataDirectory = rootDataDirectory;
+    }
+
+    Set<String> getNamespaces() {
+        return Arrays.stream(PaxosUseCase.values())
+                .filter(useCase -> useCase != PaxosUseCase.LEADER_FOR_ALL_CLIENTS)
+                .map(useCase -> useCase.logDirectoryRelativeToDataDirectory(rootDataDirectory))
+                .flatMap(DiskNamespaceLoader::getNamespacesFromUseCaseResolvedDirectory)
+                .collect(Collectors.toSet());
+    }
+
+    private static Stream<String> getNamespacesFromUseCaseResolvedDirectory(Path logDirectory) {
+        File[] directories = logDirectory.toFile().listFiles(File::isDirectory);
+        if (directories == null) {
+            LoggerFactory.getLogger(DiskNamespaceLoader.class)
+                    .error("Could not get file for the directory {}", SafeArg.of("dirName", logDirectory));
+            return Stream.of();
+        }
```

Comment:
I'd suggest we write this as: 

- if `logDirectory` does not exist, return `Stream.of()` without complaints, or maybe an INFO log
- if it does and we get null, log error and throw: this is a strange state